docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md

```diff
@@ -0,0 +1,92 @@
+
+### Stakeholders
+
+| Role/Name       | Contact        | Approval       |
+|-----------------|----------------|----------------|
+| *Eng-Manager*   |     Sasha      |                |
+| *Developers*    |     Dominik    |                |
+| *Testers*       |     Duy        |                |
+
+### Requirements
+
+[EPIC](https://github.com/input-output-hk/catalyst-voices/issues/399)  
+[CIP-30](https://cips.cardano.org/cip/CIP-30)  
+[CIP-95](https://cips.cardano.org/cip/CIP-95)  
+Testing will be focused only on web applications for the time being.  
+Focus needs to be on compliance with CIP-30 and CIP-95 specs, reliability and security especially when handling wallet interactions and message signing. UI testing is out of scope, the web application will be used only for setup purpose.  
+
+### Acceptance criteria
+
+- The package must comply with CIP-30 and CIP-95 specifications, focusing on the web.  
+- High unit test coverage to ensure reliability and security.  
+- Clear and comprehensive documentation for developers.  
+- An example web application that effectively demonstrates the package's functionalities.  
+
+### Risks
+
+Documentation is not complete yet, integration dart package might be not enough  
+
+### Tools
+
+The main tool we will use is dart integration test package. We still need to investigate if all the testing needed can be archived only with dart or if we need to use playwright for the web UI part  
+
+## ACC framework
+
+> :memo: **Note:** Use the [ACC framework](https://testing.googleblog.com/2011/09/10-minute-test-plan.html)
+> to help you define the test cases
+
+### Attributes
+
+Secure, complaint with CIP 30 and 36, reliable    
+
+### Components
+
+Web application  
+
+### Capabilities
+
+User can retrieve his wallet details  
+User can sign data  
+User can sign transactions  
+User submit transactions  
+
+## Integration tests
+
+### Integration tests strategy
+
+> :memo: **Note:** Evaluate all integrations with other functions, services etc.
+
+### Integration test cases
+
+| \#  | OBJECTIVE | STEPS | EXPECTED RESULTS | TO BE AUTOMATED |
+| --- | --------- | ----- | ---------------- | --------------- |
+| 1   | User can retrieve his wallet details |  Start application, Download wallet chrome extention, Connect wallet extention in the application, Enable wallet  | Wallet details (balance, addresses, etc.) is correctly returned | YES  |
+| 2   | User can sign data | Connect the wallet as #1, sign data | No exceptions are thrown, VkeyWitness is valid | YES | 
+| 3   | User can sign transactions | Connect the wallet as #1, sign transaction data | No exceptions are thrown, and in TransactionWitnessSet, atleast one VkeyWitness is present | YES |
```

Comment:
I would consider this as a test case group for the following reason: Transaction is a set of different fields, here's the whole list: https://github.com/input-output-hk/catalyst-CIPs/blob/x509-rbac-signing-with-cip30/CIP-XXXX/README.md#specification

We don't aim to support everything as some fields are optional and we don't need them but in general we should test some combinations of fields. One example is the transaction metadata. It's optional but we will need it in catalyst voices. Thus we should have at least two tests: for a transaction with, and without metadata.

Here are the fields that we support: https://github.com/input-output-hk/catalyst-voices/tree/feat/frontend/catalyst_voices_packages/catalyst_cardano_serialization#supported-transaction-body-fields.

From https://github.com/input-output-hk/catalyst-CIPs/blob/x509-rbac-signing-with-cip30/CIP-XXXX/README.md#specification you can read which fields are required. Try to come up with different combinations of required and optional fields.